api/authorizers/build.js

```diff
@@ -1,26 +1,59 @@
-const { User, Site } = require('../models');
-
-const authorize = (user, build) => (
-  User.findById(user.id, { include: [Site] })
-    .then((userModel) => {
-      const buildSiteId = build.site || build.Site.id;
-      const matchingUserSite = userModel.Sites.find(site => buildSiteId === site.id);
-
-      if (matchingUserSite) {
-        return Promise.resolve();
-      }
-      return Promise.reject(403);
-    }
-  )
-);
+const { User, Build } = require('../models');
 
-const findOne = (user, build) => authorize(user, build);
+const shaRegex = /^[a-f0-9]{40}/;
+const validateSha = maybeSha => shaRegex.test(maybeSha);
```

Comment:
could we also/instead put this [validator](http://docs.sequelizejs.com/manual/tutorial/models-definition.html#validations) on the `Build` model's `commitSha` field? I think if we did it there we wouldn't have to explicitly put validations elsewhere, and wouldn't accidentally miss something in the future.

Reply:
Good point, totally forgot about that aspect of sequelize. Will move it there.